cmd/run/run.go

```diff
@@ -1016,3 +1037,15 @@ func (s *ServerContext) Run(ctx context.Context, config *Config) error {
 
 	return nil
 }
+
+// injectTraceMetadata is lightweight [grpc.UnaryClientInterceptor] that injects trace metadata into outgoing grpc unary call
+// without wrapping the call in new span.
+// This method is used for proxying HTTP calls to the gRPC server implementation without adding unnecessary nesting.
+func injectTraceMetadata(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
```

Comment:
Please move this into an internal telemetry middleware package `internal/middleware/telemetry`. We'll also need to verify the streaming endpoint exhibits the same behavior. We may need to implement the [StreamClientInterceptor](https://pkg.go.dev/google.golang.org/grpc#StreamClientInterceptor)

Reply:
Files restructured and strea middleware added.